fabric-tool-attributes-v1/src/main/java/net/fabricmc/fabric/api/tool/attribute/v1/ActableAttributeHolder.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tool.attribute.v1;
+
+import com.google.common.collect.Multimap;
+
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.attribute.EntityAttributeModifier;
+import net.minecraft.item.ItemStack;
+
+/**
+ * A tool whose properties can vary based on its user.
+ */
+public interface ActableAttributeHolder extends ToolAttributeHolder {
+	/**
+	 * @param stack The stack being used.
+	 * @param actor The actor attempting to mine.
+	 * @return The mining level for this stack and actor.
```

Comment:
Mining level is rather weakly specified and IIRC has been too coarse for other similar APIs elsewhere. Maybe we can remap the range and provide some guidelines, especially for harder materials? Something like `fabric level = 8 * vanilla level` and a suggestion to continue tiers in equal steps, making the tier above diamond 8 higher?

Reply:
I'm not sure if that'd be too invasive for Fabric's taste. It should be talked over.

Reply:
I don't know how invasive that'd be. If feasible, my preferred approach here would `round(fabricLevel / 8)` when crossing into a vanilla context or the full level when both block and tool are using fabric levels.

Reply:
That could introduce inconsistent behaviour between vanilla and Fabric contexts (consider a FL 10 tool trying to mine a FL 11 block: this will fail with Fabric levels but pass when both are converted to vanilla levels).